edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest/EdgeDeploymentControllerTest.cs

```diff
@@ -0,0 +1,590 @@
+                ["80/tcp"] = default(DockerEmptyStruct)
+            };
+            var creatOptions = new CreatePodParameters(null, exposedPorts, null, null, null);
+            var controller = this.CreateDeploymentController(deviceSelector, moduleLifeCycleManager, string.Empty, string.Empty);
+            string imageName = "test-image:1";
+            KubernetesModule km1 = this.CreateKubernetesModule(moduleName, creatOptions, imageName);
+
+            moduleLifeCycleManager.SetModules(moduleName);
+            await controller.DeployModulesAsync(ModuleSet.Create(km1), ModuleSet.Empty);
+
+            V1DeploymentList currentDeployments = await this.client.ListDeployments(deviceSelector);
+            V1ServiceAccountList currentServiceAccounts = await this.client.ListServiceAccounts(deviceSelector);
+            V1ServiceList currentServices = await this.client.ListServices(deviceSelector);
+            V1PersistentVolumeClaimList currentPvcList = await this.client.ListPeristentVolumeClaims();
+            Assert.Single(currentDeployments.Items, d => d.Metadata.Name == moduleName);
+            Assert.Single(currentServiceAccounts.Items, sa => sa.Metadata.Name == moduleName);
+            Assert.Single(currentServices.Items, s => s.Metadata.Name == moduleName);
+            Assert.Empty(currentPvcList.Items);
+        }
+
+        [Fact]
+        public async Task CheckIfDeleteDeploymentIsSuccessful()
+        {
+            var moduleName = "module-a";
+            var deviceSelector = $"{Kubernetes.Constants.K8sEdgeDeviceLabel}=deviceid,{Kubernetes.Constants.K8sEdgeHubNameLabel}=hostname";
+            var moduleLifeCycleManager = new DummyModuleIdentityLifecycleManager(
+                "hostname",
+                "gatewayhostname",
+                "deviceid",
+                moduleName,
+                new ConnectionStringCredentials("connectionString"));
+            var exposedPorts = new Dictionary<string, DockerEmptyStruct>
+            {
+                ["80/tcp"] = default(DockerEmptyStruct)
+            };
+            var hostConfigs = new HostConfig
+            {
+                Mounts = new List<Mount>
+                {
+                    new Mount
+                    {
+                        Type = "volume",
+                        ReadOnly = true,
+                        Source = "pvname",
+                        Target = "/tmp/volume"
+                    }
+                }
+            };
+            var creatOptions = new CreatePodParameters(null, exposedPorts, hostConfigs, null, null);
+            var controller = this.CreateDeploymentController(deviceSelector, moduleLifeCycleManager, "pvname", "storagename");
+            string imageName = "test-image:1";
+            KubernetesModule km1 = this.CreateKubernetesModule(moduleName, creatOptions, imageName);
+            var labels = new Dictionary<string, string>
+            {
+                ["net.azure-devices.edge.deviceid"] = "deviceid",
```

Comment:
I believe we have constants for label keys.

Reply:
Also you labels creation is repeated in several tests. Should you create a helper method instead? CreateDefaultLabels maybe?